src/engines/kotor/script/functions_object.cpp

```diff
@@ -72,6 +74,58 @@ void Functions::getObjectByTag(Aurora::NWScript::FunctionContext &ctx) {
 	ctx.getReturn() = search->get();
 }
 
+void Functions::getMinOneHP(Aurora::NWScript::FunctionContext &ctx) {
+	Aurora::NWScript::Object *object = ctx.getParams()[0].getObject();
+
+	Object *kotorObject = ObjectContainer::toObject(object);
+
+	if (!kotorObject)
+		throw Common::Exception("Functions::getMinOneHP(): invalid object");
+
+	ctx.getReturn() = kotorObject->getMinOneHitPoints();
+}
+
+void Functions::setMinOneHP(Aurora::NWScript::FunctionContext &ctx) {
+	Aurora::NWScript::Object *object = ctx.getParams()[0].getObject();
+	bool enabled = ctx.getParams()[1].getInt();
+
+	Object *kotorObject = ObjectContainer::toObject(object);
+
+	if (!kotorObject)
+		throw Common::Exception("Functions::setMinOneHP(): invalid object");
+
+	kotorObject->setMinOneHitPoints(enabled);
+}
+
+void Functions::getCurrentHitPoints(Aurora::NWScript::FunctionContext &ctx) {
+	Aurora::NWScript::Object *object = ctx.getParams()[0].getObject();
+
+	Object *kotorObject = ObjectContainer::toObject(object);
+
+	ctx.getReturn() = kotorObject ? kotorObject->getCurrentHitPoints() : 0;
+}
+
+void Functions::getMaxHitPoints(Aurora::NWScript::FunctionContext &ctx) {
+	Aurora::NWScript::Object *object = ctx.getParams()[0].getObject();
+
+	Object *kotorObject = ObjectContainer::toObject(object);
+
+	ctx.getReturn() = kotorObject ? kotorObject->getMaxHitPoints() : 0;
+}
+
+void Functions::setMaxHitPoints(Aurora::NWScript::FunctionContext &ctx) {
+	Aurora::NWScript::Object *object = ctx.getParams()[0].getObject();
+	int maxHitPoints = ctx.getParams()[1].getInt();
+
+	if (maxHitPoints == 0)
+		maxHitPoints = 1;
+
+	Object *kotorObject = ObjectContainer::toObject(object);
```

Comment:
You still don't check for kotorObject != 0 before dereferencing it the next two lines.